plugin-server/src/main/ingestion-queues/ingest-event.ts

```diff
@@ -48,6 +53,20 @@ export async function ingestEvent(
                 timeoutMessage: `After 30 seconds still running ${isSnapshot ? 'onSnapshot' : 'onEvent'}`,
             }),
         ])
+
+        if (actionMatches.length > 0) {
+            await Promise.all(
+                actionMatches.map((actionMatch) =>
+                    runInstrumentedFunction({
+                        server,
+                        event: processedEvent!,
```

Comment:
Should we get rid of the `!` somehow?

Reply:
lol I can try :D

Reply:
done, this was weird.

Reply:
lol :D 

I guess you can never know if a mutable value is still there when a callback runs...